resources/qml/Sidebar.qml

```diff
@@ -214,7 +214,7 @@ Rectangle
                 anchors.left: parent.left
                 anchors.leftMargin: model.index * (settingsModeSelection.width / 2)
                 anchors.verticalCenter: parent.verticalCenter
-                width: parent.width / 2
+                width: 0.5 * parent.width - (modesListModel.get(index).showFilterButton ? toggleFilterButton.width : 0)
```

Comment:
I get that; my point is that you are already changing the button's width based on whether it should be visible or not. So the end result is the same: if you show the button, its width will be > 0 and the other two buttons resize based on the button's width. If the button is hidden, its width will be 0 and the buttons will not subtract any size.